lib/puppet/application/master.rb

```diff
@@ -295,6 +295,9 @@ def start_webrick_master
     daemon.server = Puppet::Network::Server.new(Puppet[:bindaddress], Puppet[:masterport])
     daemon.daemonize if Puppet[:daemonize]
 
+    # Setup signal traps immediately after daemonization so we clean up the daemon
+    daemon.set_signal_traps
```

Comment:
should this happen before we daemonize like we do for the agent (when `onetime` is false)?